### PR TITLE
- add gnu make

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **make/4.2.1**

(tested on [TBB](https://github.com/conan-io/conan-center-index/pull/245))

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/make/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "4.2.1":
+    url: "http://ftpmirror.gnu.org/gnu/make/make-4.2.1.tar.bz2"
```
Comment (danimtb) [inline recipes/make/all/conandata.yml]: Is this a legal source? ftpmirror?

File: recipes/make/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+
+    def test(self):
+        with tools.chdir(self.source_folder):
+            env_build = AutoToolsBuildEnvironment(self)
```
Comment (danimtb) [inline recipes/make/all/test_package/conanfile.py]: Could you test here CONAN_MAKE_PROGRAM value?

Comment (danimtb) [reply recipes/make/all/test_package/conanfile.py]: Maybe also a `which make` and check that the path is inside a conan cache? not sure that would feasible

File: recipes/make/all/conandata.yml (hunk 1 above)
Comment (SSE4) [reply recipes/make/all/conandata.yml]: absolutely: https://www.gnu.org/prep/ftp.html
```
GNU software is available on ftp.gnu.org under the directory /gnu. Please try to use one of the many mirrors of our site listed below: the mirrors will give you faster response. Some mirrors also provide a copy of the rest of ftp.gnu.org; try the parent directory if you're interested. You can use the generic url http://ftpmirror.gnu.org to automatically choose a nearby and up-to-date mirror.

```

File: recipes/make/all/test_package/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/make/all/test_package/conanfile.py]: maybe it's better to use [tools.remove_from_path](https://docs.conan.io/en/latest/reference/tools.html#tools-remove-from-path) then?

**Pull request merged**